pkgs/by-name/ne/nezha-agent/package.nix

```
@@ -1,45 +1,67 @@
 {
   lib,
   buildGoModule,
+  stdenv,
   fetchFromGitHub,
   nezha-agent,
+  versionCheckHook,
   testers,
 }:
 buildGoModule rec {
   pname = "nezha-agent";
-  version = "0.18.5";
+  version = "0.20.1";
 
   src = fetchFromGitHub {
     owner = "nezhahq";
     repo = "agent";
     rev = "refs/tags/v${version}";
-    hash = "sha256-LmWfs3aL+1lsX4ix2FjDP5g+A0wgcfziXdw5SaKlAdk=";
+    hash = "sha256-RV5xZrjbcl26qh6d+uqSVTwZgVSJ9ZZPIAfxOpUqKm8=";
   };
 
-  vendorHash = "sha256-frPAhiexFSt+KobMbf32h8xv7HMcPl5koEgSs8Nz3cs=";
+  vendorHash = "sha256-q6/265vVg6jCnDvs825nni8QFHkJpQz4xxC9MlJH2do=";
 
   ldflags = [
     "-s"
     "-w"
     "-X main.version=${version}"
+    "-X main.arch=${stdenv.hostPlatform.system}"
   ];
 
-  # The test failed due to a geoip request in the sandbox. Remove it to avoid network requirement
-  preCheck = ''
-    rm ./pkg/monitor/myip_test.go
+  checkFlags =
+    let
+      # Skip tests that require network access
+      skippedTests = [
+        "TestLookupIP"
+        "TestGeoIPApi"
+        "TestFetchGeoIP"
+        "TestCloudflareDetection"
+      ];
+    in
+    [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];
+
+  postInstall = ''
+    pushd $out/bin
+    cp agent nezha-agent
+    rm agent
+
+    # for compatibility
+    ln -s nezha-agent agent
```

```suggestion
    ln -sr nezha-agent agent
```
Relative symlinks are more reliable.